File: backend/src/main/java/com/recruitpro/model/JobSeeker.java
package com.recruitpro.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Entity
@Table(name = "job_seekers")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class JobSeeker {

    @Id
    @Column(updatable = false, nullable = false)
    private UUID id;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "avatar_url")
    private String avatarUrl;

    @Column(columnDefinition = "TEXT")
    private String bio;

    private String location;

    @Column(name = "experience_years")
    private Integer experienceYears;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "job_seeker_skills",
        joinColumns = @JoinColumn(name = "job_seeker_id"),
        inverseJoinColumns = @JoinColumn(name = "skill_id")
    )
    @Builder.Default
    private Set<Skill> skills = new HashSet<>();

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

    @PrePersist
    protected void onCreate() {
        if (id == null) id = UUID.randomUUID();
        createdAt = Instant.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = Instant.now();
    }
}

File: frontend-jobseeker/src/types/jobseeker.ts
import type { Skill } from './job';

export interface JobSeekerProfile {
  id: string;
  // Flattened user fields from backend DTO
  userId: string;
  email: string;
  fullName: string | null;
  role: string;
  status: string;
  avatarUrl: string | null;
  bio: string | null;
  location: string | null;
  experienceYears: number | null;
  skills: Skill[];
  createdAt: string;
  updatedAt: string | null;
}

export interface DashboardStats {
  jobsApplied: number;
  interviewsCount: number;
  upcomingInterviews: number;
}
